pkg/component/runtime/test_main_test.go

```diff
@@ -0,0 +1,125 @@
+package runtime
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/elastic/elastic-agent/pkg/component"
+)
+
+var (
+	fakeCompBinPath    = fakeBinaryPath("component")
+	fakeShipperBinPath = fakeBinaryPath("shipper")
+)
+
+func TestMain(m *testing.M) {
```

Comment:
Could you place this whole thing inside of a helper? The integration testing framework here uses the fake components as well, so I think we should add this in there as well.

https://github.com/elastic/elastic-agent/blob/main/testing/integration/fake_test.go